pkg/boot/kexec/genpurg.go

```diff
@@ -0,0 +1,95 @@
+//go:build ignore
+
+package main
+
+import (
+	"bytes"
+	"encoding/hex"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+)
+
+func main() {
+	var fail bool
+	f, err := os.Create("asm.go")
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer func() {
+		if err := f.Close(); err != nil {
+			log.Fatal("Closing %v: %v", f.Name(), err)
+		}
+	}()
+
+	if _, err := fmt.Fprintf(f, "package kexec\nvar purgatories = map[string]*purgatory {\n"); err != nil {
+		log.Fatal(err)
```

Comment:
If we exit with log fatal - defer func will not be called - is this intended? The same goes for the other Fatal calls here.